libs/application/ui-shell/src/reducer/reducerUtils.ts

```diff
@@ -29,53 +30,141 @@ import {
   RepeaterScreen,
 } from '../types'
 
+export const screenRequiresAnswer = (screen: FormScreen) => {
+  if (!screen.isNavigable) {
+    return false
+  }
+
+  if (screen.type === FormItemTypes.REPEATER) {
+    return true
+  } else if (screen.type === FormItemTypes.MULTI_FIELD) {
+    let screensThatRequireAnswer = 0
+
+    for (const subScreen of screen.children) {
+      if (screenRequiresAnswer(subScreen)) {
+        screensThatRequireAnswer += 1
+      }
+    }
+
+    return screensThatRequireAnswer > 0
+  }
+
+  const doesNotRequireAnswer = get(screen, 'doesNotRequireAnswer') === true
+  return !doesNotRequireAnswer
+}
+
+export const screenHasBeenAnswered = (
+  screen: FormScreen,
+  answers: FormValue,
+  checkIfPartlyAnswered = false,
+) => {
+  if (!screenRequiresAnswer(screen)) {
+    return true
+  }
+
+  if (!screen.id) {
+    return false
+  }
+
+  const answer = getValueViaPath(answers, screen.id)
+  const answerHasValue = answer !== undefined && answer !== null
+
+  if (screen.type === FormItemTypes.REPEATER) {
+    // We do not need to check all individual screens here like we do
+    // with multi field since the screens are rendered individually
+    // and will be checked individually
+    return isArray(answer) && answer.length > 0
+  } else if (screen.type === FormItemTypes.MULTI_FIELD) {
+    let numberOfAnswers = 0
+    let numberOfNotNavigable = 0
+
+    for (const subScreen of screen.children) {
+      if (!subScreen.isNavigable) {
+        numberOfNotNavigable += 1
```

Comment:
Does this make sense for multi fields? The sub fields should either be answered, or not require answer. Besides, this is checkedo again in screenHasBeenAnswered(subScreen)+screenRequiresAnswer(screen).